### PR TITLE
feat(api): Add missing information to host and service endpoints

### PR DESCRIPTION
**Pull request by vhr**
## Description

add new APIs of detail of host and services as part of the resources. The host\service objects are extended with the missing information

**Fixes** MON-5071

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

GET `/centreon/api/beta/monitoring/resources/hosts/14`
GET `/centreon/api/beta/monitoring/resources/hosts/14/services/24`

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [ x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsHost.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Centreon\Domain\Monitoring\Model\ResourceDetailsHost:
+    exclusion_policy: none
+    virtual_properties:
+        getDuration:
+            name: getDuration
+            serialized_name: duration
+            type: string
+            groups:
+                - 'resource_details_host'
+        getDowntimesAuthor:
```
Comment (kduret) [inline config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsHost.yml:10]: why it is a virtual property ?

File: config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsHost.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            name: getDowntimesAuthor
+            serialized_name: downtimes
+            type: string
+            groups:
+                - 'resource_details_host'
+        getAcknowledgementAuthor:
```
Comment (kduret) [inline config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsHost.yml:16]: why it is a virtual property ?

File: config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsService.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+Centreon\Domain\Monitoring\Model\ResourceDetailsService:
+    exclusion_policy: none
+    virtual_properties:
+        getDuration:
+            name: getDuration
+            serialized_name: duration
+            type: string
+            groups:
+                - 'resource_details_service'
+        getDowntimesAuthor:
```
Comment (kduret) [inline config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsService.yml:10]: why it is a virtual property ?

File: config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsService.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            name: getDowntimesAuthor
+            serialized_name: downtimes
+            type: string
+            groups:
+                - 'resource_details_service'
+        getAcknowledgementAuthor:
```
Comment (kduret) [inline config/packages/serializer/Centreon/Monitoring.Model.ResourceDetailsService.yml:16]: why it is a virtual property ?

File: src/Centreon/Domain/Monitoring/Model/ResourceDetailsTrait.php [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    abstract public function getAcknowledgement(): ?Acknowledgement;
+
+    /**
+     * @return string|null
+     */
+    public function getDowntimesAuthor(): ?string
+    {
+        $downtimes = $this->getDowntimes();
+        return $downtimes
+            ? (end($downtimes))->getAuthorName()
```
Comment (kduret) [inline src/Centreon/Domain/Monitoring/Model/ResourceDetailsTrait.php:68]: the return downtimes should be a list, so we should return author by downtime (no the last one)

Comment (vhr) [reply src/Centreon/Domain/Monitoring/Model/ResourceDetailsTrait.php]: it's changed right now to return the parent list of downtimes and acknowledgment

**Pull request merged**